pkgs/coverage/test/test_with_coverage_test.dart

```diff
@@ -122,6 +124,60 @@ dependency_overrides:
       ['pub', 'global', 'run', 'coverage:test_with_coverage'],
     );
   });
+
+  test(
+      'dart run bin/test_with_coverage.dart --fail-under succeeds when coverage meets threshold',
+      () async {
+    final process = await _run([
+      'run',
+      _testWithCoveragePath,
+      '--fail-under=100', // This should pass as coverage=100% when all tests run
+      '--port',
+      '${_port++}',
+    ]);
+    await process.shouldExit(0);
+  });
+  test(
+      'dart run bin/test_with_coverage.dart --fail-under fails when coverage is below threshold',
+      () async {
+    final process = await _run([
+      'run',
+      _testWithCoveragePath,
+      '--fail-under=90', // This should throw an exit(1) as coverage =27.27% when
+      '--port', // only the `sum` test is run i.e. out of 11 lines only 3 lines
+      '${_port++}', //i.e. [5,7,8]will have hits>0
+      '--',
+      '-N',
+      'sum',
+    ]);
+    await process.shouldExit(1);
+  });
+  test(
+      'dart run bin/test_with_coverage.dart -b --fail-under succeeds when coverage meets threshold',
+      () async {
+    final process = await _run([
+      'run',
+      _testWithCoveragePath,
+      '-b',
+      '--fail-under=95', // This should pass as total lines+branches covered=20
```

Comment:
Your passing case is 95% and the failing case is 96%. This is pretty brittle. If the test file changes at all it'll break. Maybe make the passing case 90% and the failing case 99% or something?